src/logging/LogMessage.cpp

```diff
@@ -37,10 +40,14 @@ const std::string& LogMessage::getMessage() const {
 }
 
 std::string LogMessage::toString() const {
-    struct std::tm result = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, nullptr};
-    if (localtime_r(&m_instant, &result) == nullptr) {
-        throw AssertionError(format("Unable to get the date."));
+    std::tm result(stdcxx::tm());
+    try {
+        boost::date_time::c_time::localtime(&m_instant, &result);
+    }
+    catch (const std::runtime_error & err) {
+        throw AssertionError(err.what());
     }
+
```

Comment:
Maybe a good design is to create a `localtime` function in `stdcxx` that return a `std::tm` structure. Then, you can remove the `stdcxx::tm()`:
```
std::tm localtime(const std::time_t& instant) {
#if defined __GNUC__
    std::tm result {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, nullptr};
#else
    std::tm result {0, 0, 0, 0, 0, 0, 0, 0, 0};
#endif

    try {
        boost::date_time::c_time::localtime(&instant, &result);
    }
    catch (const std::runtime_error & err) {
        throw AssertionError(err.what());
    }

    return result;
}
```

NB: this implementation has to be verified.

Reply:
You can merge `stdcxx::put_time` and `stdcxx::localtime` in a `stdcxx/time.hpp`